shopfloor_ui_tests/cypress/integration/shopfloor_mobile_base_auth/test_authentication.js

```diff
@@ -0,0 +1,133 @@
+describe("Test to make sure that the user can log in and log out", () => {
+    // This test works for both apikey and standard (username / password) authentication.
+    // The test detects the type of credentials expected in the login form
+    // and takes the appropriate credentials from the config file (cypress.json).
+    // This means that modules shopfloor_mobile_base_auth_user and
+    // shopfloor_mobile_base_auth_api_key can be tested with it.
+
+    describe("Log in to the Shopfloor app", () => {
+        describe("Preparation tests", () => {
+            it("Checks that unauthenticated users are redirected to the Login page", () => {
+                cy.visit(Cypress.config("baseUrl"));
+                cy.url().should("eq", Cypress.config("baseUrl") + "login");
+            });
+
+            it("Checks that the request to user_config fails (user is not authenticated)", () => {
+                cy.intercept_user_config();
+                cy.wait_for_user_config({reload: true});
+            });
+
+            it("Makes sure that the user_config or the authentication status are not in session storage", function () {
+                cy.get_session_storage("shopfloor_appconfig").should("not.exist");
+                cy.get_session_storage("shopfloor_authenticated").should("not.exist");
+                cy.get_session_storage("shopfloor_apikey").should("not.exist");
+            });
+        });
+
+        describe("Log in", () => {
+            before(() => {
+                cy.get("form").then(($form) => {
+                    if ($form.find("input[name='apikey']").length) {
```

Comment:
why can't we have a test for each of them?
Can't we mock what `shopfloor_app_info.auth_type` gives back?
Of course this will be easier when we'll have per-app auth from #316.

I think we should already separate these tests because they are not testing the same thing.

Reply:
We can (in fact, I had them as separate tests originally). However, most of the tests are actually the same (between user and apikey auth) so I went for a single test that checks what kind of authentication is expected.

Reply:
If that's acceptable, I'll separate them.